### DISCUSSION TITLE
Ubuntu 20 network install fails to mount root

### DISCUSSION THREAD

**Discussion by kai4785** [ipxe/ipxe#709] [General]
I am getting the same behavior as this Ubuntu+Grub+EFI bug.
https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1967562
https://lists.gnu.org/archive/html/grub-devel/2022-04/msg00055.html

Except I'm not chain loading to grub and then loading the kernel. I'm booting directly into the casper kernel and initrd.
![image](https://user-images.githubusercontent.com/729693/185537747-3e95c0b6-c573-4836-a4b0-eee0abf4f7a1.png)
![image](https://user-images.githubusercontent.com/729693/185537094-1803cbc1-4584-4391-8f28-6476790d3675.png)

Reproduction steps:

Download https://cdimage.ubuntu.com/jammy/daily-live/current/jammy-desktop-amd64.iso
Extract the files `/casper/{vmlinuz,initrd}` and slap them into your ipxe config, something like so:
Boot an EFI system with iPXE

```
#!ipxe

set mirror-url http://localhost
set version 22.04
set distro jammy

# sudo mount -o loop mirror/ubuntu-cd/ubuntu-mate/jammy/daily-live/current/jammy-desktop-amd64.iso /tmp/iso/
# cp /tmp/iso/casper/{vmlinuz,initrd} mirror/ipxe/ubuntu/22.04/
# sudo umount /tmp/iso
set kernel-url ${mirror-url}/ipxe/ubuntu/${version}/vmlinuz
set initrd-url ${mirror-url}/ipxe/ubuntu/${version}/initrd

imgfree
kernel ${kernel-url} root=/dev/ram0 ramdisk_size=1500000 ip=dhcp url=${mirror-url}/ubuntu-cd/ubuntu-server/${distro}/daily-live/current/${distro}-live-server-amd64.iso
initrd ${initrd-url}

boot || shell
```
The options `root` and `ramdisk_size` are optional. BIOS boots just fine with out them. You just end up with a `null` device name in the stack trace.

**Comment by NiKiZe** [comment]
You don't have any initrd= on the kernel line.
In efi mode you are required to tell the kernel which files to load as initrd.

**Comment by stappersg** [comment]
In #674 is a working _"Ubuntu from .iso"_ documented. Please report your working version also.

**Comment by kai4785** [comment]
Excellent. Thank you for the quick response. This is working for me.

```
#!ipxe

set mirror-url http://localhost
set version 22.04
set distro jammy

# sudo mount -o loop mirror/ubuntu-cd/ubuntu-mate/jammy/daily-live/current/jammy-desktop-amd64.iso /tmp/iso/
# cp /tmp/iso/casper/{vmlinuz,initrd} mirror/ipxe/ubuntu/22.04/
# sudo umount /tmp/iso
set kernel-url ${mirror-url}/ipxe/ubuntu/${version}/vmlinuz
set initrd-url ${mirror-url}/ipxe/ubuntu/${version}/initrd

imgfree
kernel ${kernel-url} initrd=initrd ip=dhcp url=${mirror-url}/ubuntu-cd/ubuntu-server/${distro}/daily-live/current/${distro}-live-server-amd64.iso
initrd ${initrd-url}

boot || shell
```
